manim/mobject/coordinate_systems.py

```diff
@@ -127,56 +171,55 @@ def input_to_graph_point(self, x, graph):
+        y_range=None,
+        x_length=config.frame_width - 2,
+        y_length=config.frame_height - 2,
         axis_config=None,
         x_axis_config=None,
         y_axis_config=None,
-        center_point=ORIGIN,
-        **kwargs
+        **kwargs,
     ):
-        CoordinateSystem.__init__(
-            self, x_min=x_min, x_max=x_max, y_min=y_min, y_max=y_max
-        )
         VGroup.__init__(self, **kwargs)
+        CoordinateSystem.__init__(self, x_range, y_range, x_length, y_length)
 
-        self.axis_config = {
-            "color": LIGHT_GREY,
-            "include_tip": True,
-            "exclude_zero_from_default_numbers": True,
-        }
-        self.x_axis_config = {"x_min": self.x_min, "x_max": self.x_max}
-        self.y_axis_config = {
-            "x_min": self.y_min,
-            "x_max": self.y_max,
-            "label_direction": LEFT,
-            "rotation": 90 * DEGREES,
-        }
+        self.axis_config = {"include_tip": True, "numbers_to_exclude": [0]}
+        self.x_axis_config = {}
+        self.y_axis_config = {"rotation": 90 * DEGREES, "label_direction": LEFT}
 
         self.update_default_configs(
             (self.axis_config, self.x_axis_config, self.y_axis_config),
             (axis_config, x_axis_config, y_axis_config),
         )
-        self.center_point = center_point
-        self.x_axis = self.create_axis(self.x_axis_config)
-        self.y_axis = self.create_axis(self.y_axis_config)
+        self.x_axis = self.create_axis(self.x_range, self.x_axis_config, self.x_length)
+        self.y_axis = self.create_axis(self.y_range, self.y_axis_config, self.y_length)
+
         # Add as a separate group in case various other
         # mobjects are added to self, as for example in
         # NumberPlane below
-        self.axes = VGroup(self.x_axis, self.y_axis, dim=self.dim)
+        self.axes = VGroup(self.x_axis, self.y_axis)
         self.add(*self.axes)
-        self.shift(self.center_point)
+        self.center()
 
     @staticmethod
     def update_default_configs(default_configs, passed_configs):
         for default_config, passed_config in zip(default_configs, passed_configs):
             if passed_config is not None:
                 update_dict_recursively(default_config, passed_config)
 
-    def create_axis(self, axis_config):
-        return NumberLine(**merge_dicts_recursively(self.axis_config, axis_config))
+    def create_axis(self, range_terms, axis_config, length):
+        new_config = merge_dicts_recursively(self.axis_config, axis_config)
+        new_config["length"] = length
+        axis = NumberLine(range_terms, **new_config)
+
+        # without the if/elif, graph does not exist when min > 0 or max < 0
+        if range_terms[0] > 0:
+            axis.shift(-axis.number_to_point(range_terms[0]))
+        elif range_terms[1] < 0:
+            axis.shift(-axis.number_to_point(range_terms[1]))
+        else:
+            axis.shift(-axis.number_to_point(0))
```

Comment:
Not really sure what's happening here, but to me it looks like this logic should be implemented in `NumberLine` itself

Reply:
I did some light testing and in the current iteration it belongs better in `create_axis`. Shifting this functionality to `NumberLine` would take some work restructuring for little benefit since this is specifcally tuned for `Axes`.